modules/gltf/src/gltf/gltf-parser.js

```diff
@@ -224,53 +165,109 @@ export default class GLTFParser {
 
   // PARSING HELPERS
 
-  _getFullUri(uri, base) {
-    if (uri.startsWith('data:') || uri.startsWith('http:') || uri.startsWith('https:')) {
-      return uri;
+  // Load linked assets
+  async _loadBuffers(options) {
+    return await Promise.all(this.gltf.buffers.map(buffer => this._loadBuffer(buffer, options)));
+  }
+
+  async _loadBuffer(buffer, options) {
+    if (buffer.uri) {
+      const fetch = options.fetch || window.fetch;
+      const uri = this._getFullUri(buffer.uri, options.uri);
+      const response = await fetch(uri);
+      const arrayBuffer = await response.arrayBuffer();
+      buffer.data = arrayBuffer;
+      buffer.uri = null;
     }
+  }
+
+  _getFullUri(uri, base) {
+    const absolute = uri.startsWith('data:') || uri.startsWith('http:') || uri.startsWith('https:');
+    return absolute ? uri : base.substr(0, base.lastIndexOf('/') + 1) + uri;
+  }
 
-    return base.substr(0, base.lastIndexOf('/') + 1) + uri;
+  _decompressMeshes() {
+    for (const mesh of this.gltf.meshes || []) {
+      // Decompress all the primitives in a mesh
+      mesh.primitives = mesh.primitives.map(primitive => this._decompressPrimitive(primitive));
+    }
   }
 
-  // Start loading linked assets
-  _loadLinkedAssets(options) {
-    const promises = [];
+  // Unpacks one mesh primitive
+  // TODO - decompression could be threaded?
+  // TODO - fallback implementation?
 
-    if (!this.glbParser) {
-      this.gltf.buffers.forEach(buffer => {
-        // TODO: handle base64 encoded files in the non-async path
-        if (buffer.uri) {
-          // TODO: Use loaders.gl readFile API so that this works on node.js as well
-          promises.push(window
-            .fetch(this._getFullUri(buffer.uri, options.uri))
-            .then(res => res.arrayBuffer())
-            .then(data => {
-              buffer.data = data;
-            }));
-        }
-      });
+  _decompressPrimitive(primitive) {
+    // eslint-disable-next-line max-len
+    // https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_draco_mesh_compression
+    const compressedMesh =
+      primitive.extensions && primitive.extensions.KHR_draco_mesh_compression;
+
+    if (!this.DracoDecoder || !compressedMesh) {
+      // TODO if no uncompressed attributes, fail
+      return primitive;
+    }
+
+    const unpackedPrimitive = {
```

Comment:
Attach back to original **primitive**.
Current logic will eliminate additional extensions and extras!!!

Reply:
> Attach back to original primitive.
> Current logic will eliminate additional extensions and extras!!!

Good points! Will address in next PR that is focused on Draco.